tests/load_FWs.py

```diff
@@ -87,36 +93,40 @@ def load_can(drive_conf, mc):
     ini_time = time.perf_counter()
     while (time.perf_counter() - ini_time) <= TIMEOUT_NEW_FW_DETECT and not detected:
         try:
-            mc.communication.connect_servo_canopen(
-                CAN_DEVICE(drive_conf["device"]),
-                drive_conf["dictionary"],
-                drive_conf["node_id"],
-                CAN_BAUDRATE(drive_conf["baudrate"]),
-                channel=drive_conf["channel"],
-            )
+            net, servo = connect_can(drive_conf)
             # Reaching this point means we are connected
             detected = True
-            logger.info("New FW detected after: {:.1f} s".format(time.perf_counter() - ini_time))
-            mc.communication.disconnect()
+            fw_version = servo.info["firmware_version"]
+            logger.info(
+                f"New FW detected ({fw_version}) after: {time.perf_counter() - ini_time:.1f} s"
+            )
+            net.disconnect_from_slave(servo)
         except Exception as e:
             # When cannot connect
             time.sleep(SLEEP_TIME_NEW_FW_DETECT)
 
     if not detected:
-        raise Exception("New FW not detected")
+        raise Exception("Could not connect to the drive. FW loading might have failed.")
```

Comment:
I think you could use a more specific Exception. For instance ILError. What do you think?

Reply:
In fact, if you don't use ILError exception, this exception won't be handled in the main function, right? Or is your intention because Jenkins will detect it, and the job building will stop?

Reply:
I just copied the code from ingenialink: I think the idea is what you mentioned, if the exception is raised, the Jenkins job is stopped.